### ISSUE TITLE
length compares to null instead of 0 in some cases

### ISSUE THREAD

**Issue by giner** [mikefarah/yq#1427]
**Describe the bug**

In some cases comparison of length of a null object with zero does not result in success

Version of yq: 4.30.1
Operating system: linux
Installed via: binary

**Command**

The command you ran:
```bash
echo '{}' | yq '.k | length == 0'                        # true
echo '{}' | yq '.k | (length == 0)'                      # true
echo '{}' | yq '(.k | length) == 0'                      # false
echo '{}' | yq '(.k | length) == null'                   # true
echo '{}' | yq '[(.k | length), (.k | length) == 0]'     # [0, true]
echo '{}' | yq '[(.k | length), (.k | length == 0)]'     # [0, true]
echo '{}' | yq '[(.k | length) == 0, (.k | length)]'     # [false, 0]
echo '{}' | yq '[(.k | length == 0), (.k | length)]'     # [true, 0]
echo '{}' | yq '[(.k | length) == null, (.k | length)]'  # [true, 0]
```

The results of the same for jq:
```bash
echo '{}' | jq '.k | length == 0'                        # true
echo '{}' | jq '.k | (length == 0)'                      # true
echo '{}' | jq '(.k | length) == 0'                      # true
echo '{}' | jq '(.k | length) == null'                   # false
echo '{}' | jq '[(.k | length), (.k | length) == 0]'     # [0, true]
echo '{}' | jq '[(.k | length), (.k | length == 0)]'     # [0, true]
echo '{}' | jq '[(.k | length) == 0, (.k | length)]'     # [true, 0]
echo '{}' | jq '[(.k | length == 0), (.k | length)]'     # [true, 0]
echo '{}' | jq '[(.k | length) == null, (.k | length)]'  # [false, 0]
```

**Actual behavior**

Some comparisons to 0 above result in "false"

**Expected behavior**

All comparisons to 0 above result in "true"

**Comment by mikefarah** [comment]
Interesting, thanks for raising - I think I know the cause, just need to make sure the fix doesn't cause other issues...

**Comment by mikefarah** [comment]
Fixed in 4.30.3
